shared/common-adapters/plain-input.stories.js

```diff
@@ -17,6 +19,138 @@ const commonProps = {
   style: {borderWidth: 1, borderStyle: 'solid', borderColor: globalColors.black_10},
 }
 
+type TestInputState = {
+  value: string,
+}
+class TestInput extends React.Component<{}, TestInputState> {
+  state = {value: ''}
+  // prettier-ignore
+  _input = React.createRef<typeof PlainInput>()
+
+  _insertText = (t: string) => {
+    const input = this._input.current
+    if (input) {
+      const selection = input.getSelection()
+      if (selection) {
+        this.setState(
+          s => {
+            const value = s.value.substring(0, selection.start) + t + s.value.substring(selection.start)
+            return {value}
+          },
+          () => {
```

Comment:
ah, interesting. I was gonna ask how this would work with redux, but it looks like most (all?) of our inputs end up setting the state in some higher component anyway, so this would work!

Reply:
Yeah I think having the value come directly from redux risks a bad ux, since there could be some latency involved and the keypress should be very responsive. So fortunately the ux and API interests align in this case